src/ucp/wireup/address.c

```diff
@@ -486,15 +499,25 @@ ucp_address_packed_size(ucp_worker_h worker,
     } else {
         for (dev = devices; dev < (devices + num_devices); ++dev) {
             /* device md_index */
-            md_index = worker->context->tl_rscs[dev->rsc_index].md_index;
+            md_index   = worker->context->tl_rscs[dev->rsc_index].md_index;
             /* md index (+flags) can take 2 bytes with address version 2 */
-            size    += ucp_address_packed_value_size(md_index, md_mask,
-                                                     addr_version);
+            value_size = ucp_address_packed_value_size(md_index, md_mask,
+                                                       addr_version);
+            if (value_size < 0) {
+                return -1;
+            }
+
+            size      += value_size;
             if (pack_flags & UCP_ADDRESS_PACK_FLAG_DEVICE_ADDR) {
                 /* device address length */
-                size += ucp_address_packed_value_size(
-                            dev->dev_addr_len, UCP_ADDRESS_DEVICE_LEN_MASK,
-                            addr_version);
+                value_size = ucp_address_packed_value_size(
+                                dev->dev_addr_len, UCP_ADDRESS_DEVICE_LEN_MASK,
+                                addr_version);
+                if (value_size < 0) {
+                    return -1;
```

Comment:
can we return "value_size" to preserver the original error code?

Reply:
fixed